hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java

```diff
@@ -233,7 +235,8 @@ private void handlePartialStart(HttpPostCommand command) throws UnsupportedEncod
             if (!(groupConfig.getName().equals(groupName) && groupConfig.getPassword().equals(groupPass))) {
                 res = res.replace("${STATUS}", "forbidden");
             } else {
-                boolean success = node.getNodeExtension().triggerPartialStart();
+                final InternalHotRestartService hotRestartService = node.getNodeExtension().getInternalHotRestartService();
+                boolean success = hotRestartService != null && hotRestartService.triggerPartialStart();
```

Comment:
Would it not be an idea to always have a hotRestartService but in case of not being enabled, that a NullHotRestartService is used? This way you never need to do null checks.

http://refactoring.com/catalog/introduceNullObject.html